namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -99,40 +99,39 @@ object HttpControlService {
     val prefix = "/api/1/dtabs"
     val prefixSlash = s"$prefix/"
 
-    def unapply(path: String): Option[Option[String]] =
-      if (path == prefix) Some(None)
-      else if (!path.startsWith(prefixSlash)) None
-      else path.stripPrefix(prefixSlash) match {
-        case "" => Some(None)
-        case ns => Some(Some(ns))
+    def unapply(request: Request): Option[(Method, Option[Ns])] =
+      if (request.path == prefix) Some(request.method, None)
+      else if (!request.path.startsWith(prefixSlash)) None
+      else request.path.stripPrefix(prefixSlash) match {
+        case "" => Some(request.method, None)
+        case ns => Some(request.method, Some(ns))
       }
   }
 
   trait NsPathUri {
     val prefix: String
 
-    def unapply(uri: String): Option[(Ns, Path)] = {
-      if (uri.startsWith(prefix)) {
-        uri.stripPrefix(prefix).split("/").toSeq match {
-          case ns +: path if path.nonEmpty => Some((ns, Path.Utf8(path: _*)))
-          case _ => None
-        }
+    def unapply(request: Request): Option[(Ns, Path)] = {
+      if (request.path.startsWith(prefix)) {
+        val ns = request.path.stripPrefix(prefix)
+        val path = Path.read(request.getParam("path"))
```

Comment:
this will blow up if the param is not a valid path -- does that cause a 400? (test please?)